roxie/roxiemem/roxiemem.cpp

```diff
@@ -1767,11 +1791,15 @@ class PackedFixedSizeHeaplet : public ChunkedHeaplet
 
         //If count == 1 then no other thread can be releasing at the same time - so avoid locked operation
         //Subtract 1 here to try and minimize the conditional branches
-        unsigned rowCount = atomic_read(&header->count)-1;
+        unsigned rowCount = header->count.load(std::memory_order_relaxed)-1;
+
+        //NORE: See comment on FixedSizeHeaplet::noteReleased() regarding the memory order operands
+
         //No need to reassign to rowCount if dec and read is required - since only top bit is used, and that must be the same
         //No need to ensure header->count is 0 because the free list overlaps the count so it is never checked.
-        if (ROWCOUNT(rowCount) == 0 || ROWCOUNT(atomic_dec_and_read(&header->count)) == 0)
+        if (ROWCOUNT(rowCount) == 0 || ROWCOUNT(header->count.fetch_sub(1, std::memory_order_release)) == 1)
```

Comment:
I think it's the difference between atomic_dec_and_test versus atomic_dec_and_read that is confusing me